### PR TITLE
[CT-1327] place post only orders first in prepare check state

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method `IsPostOnlyOrder` to determine if an order is post-only.
	- Enhanced order placement logic with a new parameter `postOnlyFilter` for conditional processing of orders.
	- Updated replay operations to conditionally skip non-post-only orders based on the new parameter.
	- Added new long-term orders with a "post-only" time-in-force option.

- **Bug Fixes**
	- Improved error handling and telemetry for order placement and removal processes.

- **Tests**
	- Updated test cases to cover new functionality related to post-only order processing and ensure robustness against edge cases.
	- Expanded tests for conditional order removals to reflect new business logic.

### CODE REVIEW

**Review by roy-dydx — APPROVED** [review]

File: protocol/x/clob/keeper/orders.go [hunk 1]
```diff
@@ -521,6 +522,12 @@ func (k Keeper) PlaceStatefulOrdersFromLastBlock(
 		}
 
 		order := orderPlacement.GetOrder()
+
+		// Prioritize placing post-only orders if the flag is set.
```
Comment (teddyding) [inline protocol/x/clob/keeper/orders.go:526]: Nit:
```suggestion
		// Only place post-only orders if the flag is set.
```

File: protocol/x/clob/keeper/orders.go [hunk 2]
```diff
@@ -521,6 +522,12 @@ func (k Keeper) PlaceStatefulOrdersFromLastBlock(
...
+		if onlyPlacePostOnly && !order.IsPostOnlyOrder() {
```
Comment (teddyding) [inline protocol/x/clob/keeper/orders.go:527]: We call this function twice, second time with `onlyPlacePostOnly` = false, which doesn't exclude the PO stateful orders. Does that mean PO orders are placed twice and is that an issue?

**Review by teddyding — DISMISSED** [review]

File: protocol/x/clob/keeper/orders.go [hunk 3]
```diff
@@ -496,6 +496,7 @@ func (k Keeper) PlaceStatefulOrdersFromLastBlock(
 	ctx sdk.Context,
 	placedStatefulOrderIds []types.OrderId,
 	existingOffchainUpdates *types.OffchainUpdates,
+	onlyPlacePostOnly bool,
```
Comment (teddyding) [inline protocol/x/clob/keeper/orders.go:499]: Nit: current naming might suggest that if `false`, the function processes ALL orders including post-only. How abt sth like `postOnlyFilter` and comment explicitly what this var does?

File: protocol/x/clob/keeper/orders.go [hunk 4]
```diff
@@ -521,6 +522,12 @@ func (k Keeper) PlaceStatefulOrdersFromLastBlock(
 		}
 
 		order := orderPlacement.GetOrder()
+
+		// Skip the order if it is a post-only order and we are only placing post-only orders.
```
Comment (teddyding) [inline protocol/x/clob/keeper/orders.go:526]: Nit: comment could be more accurate/complete - right now it just describes one of the two scenarios

**Review by teddyding — APPROVED** [review]
